csrc/ops/composite.cpp

```diff
@@ -64,32 +64,40 @@ static TensorView* newForLinear(
   auto input_domain = TensorDomain::noReductions(input->getLogicalDomain());
   auto weight_domain = TensorDomain::noReductions(weight->getLogicalDomain());
 
+  // Output has a reduction axis rK if K is not bcast
```

Comment:
I think we could insert a check that `input_domain.back()->isBroadcast() == weight_domain.back()->isBroadcast()` (similar for the other ops). h/t @jjsjann123.